service/src/main/java/com/networknt/service/SingletonServiceFactory.java

```diff
@@ -109,6 +115,13 @@ private static Object handleSingleImpl(List<String> interfaceClasses, List<Objec
                 }
             } else if(mapOrList instanceof List){
                 obj = ServiceUtil.constructByParameterizedConstructor(implClass, (List)mapOrList);
+
+                // use instance factory method to instantiate object if one was specified
+                if(initMethodName != null) {
+                    Method method = obj.getClass().getMethod(initMethodName);
+                    obj = method.invoke(obj);
```

Comment:
Will the error message that results of the provided method name not belonging to the class be understandable?